app/gui/src/model/suggestion_database/entry.rs

```diff
@@ -54,6 +54,22 @@ pub struct MissingThisOnMethod(pub String);
 
 
 
+// =============================
+// === QualifiedNameSegments ===
+// =============================
+
+/// A single segment of a fully qualifed name of an [`Entry`].
+pub type NameSegment = ImString;
+
+/// Segments of a fully qualified name of an [`Entry`].
+///
+/// A fully qualified name can be built by joining the segments using an
+/// [`ast::opr::predefined::ACCESS`] character.
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct QualifiedNameSegments(pub Vec<NameSegment>);
+
```

Comment:
Many questions here:

1. `NameSegment` is not a new type. Why? Alternatively, why `QualifiedNameSegments` is a new type? (One of this decisions is inconsistent).
2. `QualifiedNameSegments` contains list of `NameSegment`. So, `NameSegment` does not need to be `QualifiedNameSegment`? 
3. I believe that the whole `QualifiedNameSegments` name describes its structure rather than semantics. Would `QualifiedName` be just better here?